code/modules/SCP/scp216.dm

```diff
@@ -0,0 +1,259 @@
+//		/obj/item/clothing/under/rank/ntwork/nanotrasen = 400,
+//		/obj/item/clothing/under/suit_jacket/corp/nanotrasen = 400,
+//		/obj/item/clothing/under/rank/guard/heph = 400,
+//		/obj/item/clothing/under/rank/scientist/zeng = 400,
+//		/obj/item/clothing/under/rank/scientist/executive/zeng = 400,
+		/obj/item/clothing/under/pants/classicjeans = 400,
+		/obj/item/clothing/under/pants/blackjeans = 400,
+		/obj/item/clothing/under/pants/jeans = 400,
+		/obj/item/clothing/under/pants/youngfolksjeans = 400,
+//		/obj/item/clothing/under/casual_pants/baggy/track = 400,
+		/obj/item/clothing/under/pants/camo = 350,
+//		/obj/item/clothing/under/rank/security = 300,
+//		/obj/item/clothing/under/rank/dispatch = 300,
+//		/obj/item/clothing/under/rank/warden = 250,
+//		/obj/item/clothing/under/rank/warden/corp = 250,
+		/obj/item/clothing/under/syndicate = 100,
+		/obj/item/clothing/under/syndicate/combat = 100,
+//		/obj/item/clothing/under/syndicate/terragov = 100,
+		/obj/item/clothing/suit/armor = 500,
+//		/obj/item/clothing/suit/armor/pcarrier/medium = 500,
+		/obj/item/clothing/suit/armor/riot = 450,
+		/obj/item/clothing/suit/armor/bulletproof = 450,
+		/obj/item/clothing/suit/armor/laserproof = 450,
+		/obj/item/clothing/suit/armor/vest/warden = 450,
+//		/obj/item/clothing/suit/armor/swat/officer = 400,
+//		/obj/item/clothing/suit/armor/vest/ert = 300,
+//		/obj/item/clothing/suit/armor/vest/ert/security = 300,
+//		/obj/item/clothing/suit/storage/vest/tactical = 250,
+//		/obj/item/clothing/suit/storage/vest/merc = 250,
+//		/obj/item/clothing/suit/armor/pcarrier/tan/tactical = 250,
+//		/obj/item/excalibur = 20,
+//		/obj/item/gun/projectile/pistol = 80,
+//		/obj/item/gun/projectile/pistol/military = 60,
+//		/obj/item/gun/projectile/automatic = 30,
+//		/obj/item/gun/projectile/heavysniper = 5,
+		/obj/item/market_uplink = 60,
+		/obj/item/multitool/uplink = 60,
+		/obj/item/soulstone = 50,
+		/obj/item/chameleon = 30,
+		/obj/item/assembly/flash/handheld = 20,
+//		/obj/item/reagent_containers/pill/scp500 = 10,
+//		/obj/item/storage/pill_bottle/scp500 = 1,
+		/mob/living/simple_animal/mouse/white = 150,
+		/mob/living/simple_animal/pet/dog/corgi = 100,
+		/mob/living/simple_animal/slime = 80,
+		/mob/living/simple_animal/hostile/carp = 50,
+		/mob/living/simple_animal/hostile/asteroid/wolf = 40,
+		/mob/living/simple_animal/hostile/asteroid/polarbear
+		)
+
+/obj/structure/scp216/Initialize()
+	. = ..()
+	SCP = new /datum/scp(
+		src, // Ref to actual SCP atom
+		"safe", //Name
+		SCP_SAFE, //Obj Class
+		"216", //Numerical Designation
+	)
+
+/obj/structure/scp216/Destroy()
+	LAZYCLEARLIST(all_codes) // Forever gone
+	return ..()
+
+/obj/structure/scp216/update_icon()
+	..()
+	if(open)
+		icon_state = "[initial(icon_state)]-open"
+	else
+		icon_state = initial(icon_state)
+
+/obj/structure/scp216/attackby(obj/item/I, mob/user)
+	if(!open)
+		return ..()
+	InsertItem(user, I, current_code)
```

Comment:
attackby -> `item_interaction()`, see other uses of `item_interaction()` for how to do this, or ask me on discord.

Reply:
Does not exist on our fork yet. Note for Self: TD with a broader pull over from Daed.